src/Illuminate/Database/Eloquent/Prunable.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Illuminate\Database\Eloquent;
+
+use Illuminate\Database\Events\ModelsPruned;
+use LogicException;
+
+trait Prunable
+{
+    /**
+     * Prune all prunable models in the database.
+     *
+     * @param  int  $chunkSize
+     * @return int
+     */
+    public function pruneAll(int $chunkSize = 1000)
+    {
+        $total = 0;
+
+        $this->prunable()
+            ->when(in_array(SoftDeletes::class, class_uses_recursive(get_class($this))), function ($query) {
+                $query->withTrashed();
+            })->chunkById($chunkSize, function ($models) use (&$total) {
+                $models->each->prune();
+
+                $total += $models->count();
+
+                event(new ModelsPruned(static::class, $total));
+            });
+
+        return $total;
+    }
+
+    /**
+     * Get the prunable model query.
+     *
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function prunable()
```

Comment:
This method could be defined as an abstract method. @nunomaduro 
If a pull request is desired, I would take care of it. 😄

Reply:
@lairg99 No plans for that.

Reply:
Purely out of interest, why?